core/domain/classifier_services.py

```diff
@@ -118,3 +122,74 @@ def classify_string_classifier_rule(state, normalized_answer):
             return None
 
     return None
+
+
+def get_classifier_from_model(classifier_model):
+    """Returns a classifier domian object given a classifier model loaded
+    from datastore.
+    """
+    return classifier_domain.Classifier(
+        classifier_model.id, classifier_model.exp_id,
+        classifier_model.exp_version_when_created,
+        classifier_model.state_name, classifier_model.algorithm_id,
+        classifier_model.cached_classifier_data,
+        classifier_model.data_schema_version)
+
+def get_classifier_by_id(classifier_id):
+    """Returns a classifier domain object given a classifier id.
+    """
+    classifier_model = classifier_models.ClassifierModel.get(
+        classifier_id, strict=False)
+    if classifier_model is None:
+        raise Exception("No classifier found for the classifer's id.")
+    else:
+        classifier = get_classifier_from_model(classifier_model)
+        return classifier
+
+
+def _create_classifier(classifier):
+    """Creates classifier model in the datastoer given a classifier
+       domain object.
+    """
+    classifier_models.ClassifierModel.create(
+        classifier.exp_id, classifier.exp_version_when_created,
+        classifier.state_name, classifier.algorithm_id,
+        classifier.cached_classifier_data, classifier.data_schema_version,
+        )
+
+
+def _save_classifier(classifier_model, classifier):
```

Comment:
I don't have a particular use case in mind. I just wanted to confirm that we aren't worried about this, since I noticed the possibility of state_name not being part of the version of the exploration corresponding to exp_version_when_created. It might be worth explicitly adding a comment in the code stating that this mismatch might occur due to subsequent renamings of the state.